luxonis_train/core/utils/infer_utils.py

```diff
@@ -27,13 +33,93 @@ def render_visualizations(
                     name = name.replace("/", "_")
                     cv2.imwrite(str(save_dir / f"{name}_{i}.png"), viz_arr)
                     i += 1
-                else:
+                elif show:
                     cv2.imshow(name, viz_arr)
+                else:
+                    rendered_visualizations[name].append(viz_arr)
 
-    if save_dir is None:
+    if save_dir is None and show:
         if cv2.waitKey(0) == ord("q"):
             exit()
 
+    return rendered_visualizations
+
+
+def prepare_and_infer_image(model, img: np.ndarray, labels: dict, view: str):
+    """Prepares the image for inference and runs the model."""
+    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
```

Comment:
This part actually depends if `trainer.preprocessing.train_rgb` is set to True or False in the config (True by default - means that we should feed RGB in the model).

Reply:
If I am not mistaken, this should be handled in the augmentations which take RGB images.

Reply:
Oh you, you are right.